tensorflow/contrib/image/python/ops/image_ops.py

```diff
@@ -122,6 +147,40 @@ def angles_to_projective_transforms(angles, image_height, image_width):
       axis=1)
 
 
+def translations_to_projective_transforms(translations):
+  """Returns projective transform(s) for the given translation(s).
+
+  Args:
+      translations: A 2-element list representing [dx, dy] or a vector of
+          2-element lists representing [dx, dy] to translate for each image
+          (for a batch of images).
+
+  Returns:
+      A tensor of shape (num_images, 8) projective transforms which can be given
+          to `tf.contrib.image.transform`.
+  """
+  translation_or_translations = ops.convert_to_tensor(
+      translations, name="translations", dtype=dtypes.float32)
+  if len(translation_or_translations.get_shape()) == 1:
```

Comment:
what if .get_shape() returns TensorShape(None)?

Reply:
That is an issue throughout, but it's really useful to do this validation eagerly and be able to return a tensor with the correct shape (HW, HWC, or NHWC). Added docstrings indicating which arguments need a static shape with a known rank.